### PR TITLE
feat(VerticalMenu): adds component

### PR DESCRIPTION
**Pull request by riyalohia**
### What does this implement/fix? Explain your changes.

...

### Does this close any currently open issues?

...

### Any other comments?

...

### Dependent PRs/Commits

...

### Describe breaking changes, if any.

...

### Checklist

Check all those that are applicable and complete.

- [ ] Merged with latest `master` branch

### CODE REVIEW

File: core/components/organisms/navigation/VerticalNavigation.tsx [hunk 1]
```diff
@@ -128,100 +128,186 @@ export const VerticalNavigation = (props: VerticalNavigationProps) => {
...
+        ['Navigation-pills']: true,
+        ['Navigation-pills--disabled']: menu.disabled
+      });
+
+      return (
+        <Pills
+          subtle={menu.disabled}
+          className={PillsClass}
+          appearance={getPillsAppearance(activeMenu)}
+          data-test="DesignSystem-HorizontalNavigation--Pills"
```
Comment (satyamyadav) [inline core/components/organisms/navigation/VerticalNavigation.tsx]: Please update test id of the component.

File: core/components/organisms/navigation/VerticalNavigation.tsx [hunk 2]
```diff
@@ -128,100 +128,186 @@ export const VerticalNavigation = (props: VerticalNavigationProps) => {
...
+          onClick={() => onClickHandler(menu)}
+        >
+          <Icon
+            name={menu.icon}
+            appearance={getIconAppearance(menuActive, menu.disabled)}
+          />
+        </div>
+        {menuState[menu.name] &&
+          menu.subMenu &&
+          menu.subMenu.map((subMenu, ind) => {
```
Comment (satyamyadav) [inline core/components/organisms/navigation/VerticalNavigation.tsx]: Please update index param to `smIndex`.

File: css/src/components/collapsible.css [hunk 3]
```diff
@@ -22,7 +23,7 @@
 .Collapsible-body {
   flex-grow: 1;
   overflow-y: auto;
-  transition: width 0.1s ease-in-out;
+  /* transition: width 0.1s ease-in-out; */
```
Comment (aditya-kumawat) [inline css/src/components/collapsible.css]: Please remove comment.

File: core/components/organisms/verticalMenu/MenuItem.tsx [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+          appearance="subtle"
+        />
+      );
+    }
+
+    if (menu.count !== undefined) {
+      const count = menu.count > 99 ? '99+' : menu.count;
+      const PillsClass = classNames({
+        ['VerticalMenu-count']: true,
+        ['VerticalMenu-count--disabled']: menu.disabled
```
Comment (aditya-kumawat) [inline core/components/organisms/verticalMenu/MenuItem.tsx]: Please update classname block to `MenuItem` from `VerticalMenu`.

File: core/components/organisms/verticalMenu/VerticalMenu.tsx [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+      } else {
+        if (autoCollapse) {
+          if (!expanded) setMenuState({});
+        }
+      }
+    }
+  };
+
+  const onClickHandler = (menu: Menu) => {
+    if (!menu.disabled) {
```
Comment (aditya-kumawat) [inline core/components/organisms/verticalMenu/VerticalMenu.tsx]: Not required if we are already adding `pointer-events: none` in css. Please verify.

File: core/components/organisms/verticalMenu/__stories__/index.story.tsx [hunk 6]
```diff
@@ -0,0 +1,201 @@
+import * as React from 'react';
+import { VerticalMenu } from '@/index';
+import { action } from '@storybook/addon-actions';
+import { boolean } from '@storybook/addon-knobs';
+import { Menu } from '../../Navigation';
+
+export const sideNav = () => {
```
Comment (aditya-kumawat) [inline core/components/organisms/verticalMenu/__stories__/index.story.tsx]: Please update story name to 'All'

File: core/components/organisms/verticalMenu/__stories__/variants/Rounded.story.tsx [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+  });
+
+  const onClickHandler = (menu: Menu) => {
+    action(`menu-clicked: ${JSON.stringify(menu)}`)();
+
+    setActive(menu);
+  };
+
+  return (
+    <div style={{ height: 'calc(80vh)', background: 'var(--secondary-lightest)' }}>
```
Comment (aditya-kumawat) [inline core/components/organisms/verticalMenu/__stories__/variants/Rounded.story.tsx:87]: Please remove div wrapper.

File: core/components/patterns/verticalNavigation/sideNav.story.tsx [hunk 8]
```diff
@@ -0,0 +1,111 @@
+import * as React from 'react';
+
+export const sideNav = () => <></>;
+
+const customCode = `() => {
+  const data = [
```
Comment (aditya-kumawat) [inline core/components/patterns/verticalNavigation/sideNav.story.tsx]: Please rename `data` to `menus`.

File: css/src/components/verticalMenu.css [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+
+.VerticalMenu--expanded {
+  width: var(--spacing-9);
+}
+
+.VerticalMenu-section {
+  text-transform: uppercase;
+  padding-left: var(--spacing-2);
+  padding-top: 14px;
+  padding-bottom: 6px;
```
Comment (aditya-kumawat) [inline css/src/components/verticalMenu.css]: Please use pre-defined spacing variables which are close to the corresponding values.

**Review by satyamyadav — APPROVED** [review]

**Pull request merged**